module/zfs/arc.c

```diff
@@ -311,6 +311,11 @@
 boolean_t arc_watch = B_FALSE;
 #endif
 
+/* Dirty data threshold if not overridden by the module parameter */
+#define ZFS_DIRTY_MAX_LIMIT		(4ULL * 1024 * 1024 * 1024)
```

Comment:
what is the science behind this value? I have seen it being too small for high performance systems.

Reply:
This is not something I introduced..capping it to 4GB is legacy code. I just introduced a macro here. Even for our systems we an go as high as 96GB but more the dirty data also has the risk of losing so much of it during a powercycle or panic event (unless there is a fast media acting as a log device)